### PR TITLE
Adds canMakePayments method

### PR DESCRIPTION
**Pull request by beylmk**
Will need to update to remove the includeBuilds and point to new purchases-hybrid-common

### CODE REVIEW

File: android/build.gradle
Comment (vegaro) [inline android/build.gradle:12]: Can you extract this update, this one https://github.com/RevenueCat/purchases-flutter/pull/193/files#diff-90156c7b6ee416a97411a15877e2e1eac7b5c7c630ead570c421caf5153362d4R8 and https://github.com/RevenueCat/purchases-flutter/pull/193/files#diff-38561c91b2badbeee265859a611c25856b5e46a13aaf19e30d2adb6f859b9745R6 to a separate PR? That way changes are more isolated since this is not really part of the `canMakePayments` support and if they cause issues, it's easier to revert and spot what's causing problems.

**Review by vegaro — CHANGES_REQUESTED** [review]

File: android/build.gradle
Comment (aboedo) [reply android/build.gradle:12]: I believe this part is needed for tests to pass locally, though, right? 
Agreed with separating it as a PR, but I definitely love that we're finally taking care of this. I've been updating locally and then not committing the changes for a while, which is not a great way to do things.

File: android/src/main/java/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.java
Comment (aboedo) [inline android/src/main/java/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.java:555]: we're already importing `@Nullable` in line 7, right? can we just use `@Nullable` here? I believe we do that in other places

File: revenuecat_examples/purchase_tester/android/settings.gradle
Comment (aboedo) [inline revenuecat_examples/purchase_tester/android/settings.gradle:23]: oops

File: android/build.gradle
Comment (vegaro) [reply android/build.gradle:12]: I also confirmed yesterday and it looks like this is the default when creating a new Flutter plugin. So yes, thanks for updating these 🙌🏻

File: revenuecat_examples/purchase_tester/android/settings.gradle
Comment (beylmk) [reply revenuecat_examples/purchase_tester/android/settings.gradle:23]: yeah `hybrid-common` needs to be updated, i left this here for now in case yall wanted to test?

File: ios/Classes/PurchasesFlutterPlugin.m
Comment (aboedo) [inline ios/Classes/PurchasesFlutterPlugin.m:150]: I didn't catch it when we were doing the `purchases-hybrid-common`, but I think best practice here would have been to define the type as `NSArray <NSString *>*`. 
It doesn't really matter for now, though, since the array isn't necessary in the first place.

File: lib/purchases_flutter.dart
Comment (aboedo) [inline lib/purchases_flutter.dart:600]: convention [for enum values is lowerCamelCase](https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names)

File: revenuecat_examples/purchase_tester/android/settings.gradle
Comment (aboedo) [reply revenuecat_examples/purchase_tester/android/settings.gradle:23]: oh, ok, that's fine, then. 

I'd usually recommend leaving a `// TODO: remove before merging` or a comment in the PR to make the intent clear

File: test/purchases_flutter_test.dart
Comment (aboedo) [inline test/purchases_flutter_test.dart:91]: 🤔 
I'd also check that the value is passed along correctly, and have the response be random so that if for some reason we make a change that makes the code only compatible with `true`, it gets caught: 

```dart
response = randomGenerator.nextBool();
var receivedCanMakePayments = await Purchases.canMakePayments();
expect(receivedCanMakePayments, response);
```
that way if we later broke [this line](https://github.com/RevenueCat/purchases-flutter/pull/193/files#diff-50113335b625b38d77e98819f82d0437574295a701bcabd0450f59d5f7f12507R522
) tests would fail

**Review by aboedo — COMMENTED** [review]
looks great so far, left a couple of minor comments

File: test/purchases_flutter_test.dart
Comment (beylmk) [reply test/purchases_flutter_test.dart:91]: i passed a parameter in the second test, were you thinking of something else?

good call on the random 👍

File: ios/Classes/PurchasesFlutterPlugin.m
Comment (beylmk) [reply ios/Classes/PurchasesFlutterPlugin.m:150]: well that one isn't merged yet either :)

**Comment by beylmk** [comment]
@aboedo @vegaro i think this is ready now! updated to `purchases-hybrid-common` 1.7.0 and removed local build stuff

File: ios/Classes/PurchasesFlutterPlugin.m
Comment (aboedo) [inline ios/Classes/PurchasesFlutterPlugin.m:441]: we should name this one `canMakePaymentsWithFeatures:result:` to follow the obj-c practice of referencing the first param in the method name

**Review by aboedo — APPROVED** [review]
looks great! good to go once `purchases-hybrid-common` 1.7.1 is out and we update

File: android/build.gradle
Comment (aboedo) [inline android/build.gradle:41]: we can bump this to 1.7.1 now

**Review by aboedo — APPROVED** [review]
looks good to go! just a minor naming update and bump to 1.7.1

**Comment by beylmk** [comment]
@vegaro when you get a chance to re-review :)

File: revenuecat_examples/purchase_tester/lib/main.dart
Comment (vegaro) [inline revenuecat_examples/purchase_tester/lib/main.dart:32]: There's some stuff to be reverted in this file 😃

**Review by vegaro — CHANGES_REQUESTED** [review]

**Review by vegaro — APPROVED** [review]

**Pull request merged**
